### PR TITLE
[MTM-58365]: Adding advantage of certificate inference in obtaining device access token

### PR DESCRIPTION
**Pull request by cumulocitypans**
https://cumulocity.atlassian.net/browse/MTM-58365

### CODE REVIEW

**Review by mbak-c8y — APPROVED** [review]

File: content/device-integration/device-integration-rest-bundle/device-authentication.md [hunk 1]
```diff
@@ -13,16 +13,12 @@ Mutual TLS (mTLS) is a security protocol that uses X.509 certificates for both c
 The mTLS protocol is commonly used to secure connections in web services, APIs, and other networked applications. When generating tokens using mTLS, the process involves the authentication of both the client and the server using X.509 certificates.
 
 Retrieving device access tokens from the platform with certificates does not require the tenant ID, username and password. Authentication information will be obtained from the certificates.
+Immediate issuer of client certificate must present in Platform's truststore, if not then whole certificate chain needs to send in header and root or any intermediate certificate must be present in the Platform's truststore.
```
Comment (BeateRixen) [inline content/device-integration/device-integration-rest-bundle/device-authentication.md:16]:
```suggestion
An immediate issuer of the client certificate must be provided in the platform's truststore. If this is not the case, then the whole certificate chain must send in the header and root or any intermediate certificate must be available in the platform's truststore.
```

File: content/device-integration/rest-client-examples-bundle/hello-rest-x509.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ The following configuration is required before calling the device access token A
 * PLATFORM_URL - The URL of the platform.
 * PLATFORM_MTLS_PORT - Port 8443 is available for device access token API
 * DEVICE_ACCESS_TOKEN_PATH API - The endpoint responsible for the mTLS protocol.
-* LOCAL_DEVICE_CHAIN - The whole chain in PEM format.
+* LOCAL_DEVICE_CHAIN - The whole chain in PEM format. This header is not mandatory, if we have immediate issuer of device certificate in {{< product-c8y-iot >}}.
```
Comment (BeateRixen) [inline content/device-integration/rest-client-examples-bundle/hello-rest-x509.md:55]:
```suggestion
* LOCAL_DEVICE_CHAIN - The whole chain in PEM format. This header is not mandatory, if you have an immediate issuer of the device certificate in {{< product-c8y-iot >}}.
```

**Comment by BeateRixen** [comment]
@cumulocitypans Not sure if it is "an immediate issuer" or "the immediate issuer". Is there more than one? I used "an" in 3 places, please correct if this is wrong.

**Comment by cumulocitypans** [comment]
> @cumulocitypans Not sure if it is "an immediate issuer" or "the immediate issuer". Is there more than one? I used "an" in 3 places, please correct if this is wrong.

@BeateRixen There will be an only one immediate issuer of device certificate.

File: content/device-integration/device-integration-rest-bundle/device-authentication.md [hunk 3]
```diff
@@ -31,7 +27,12 @@ The device access token API can be called by executing the following curl statem
        -X POST \
        https://<{{< product-c8y-iot >}} tenant domain>:8443/devicecontrol/deviceAccessToken
 
-Replace `<device certificate chain>` with your valid certificate chain when registering with {{< product-c8y-iot >}}.
+Replace `<device certificate chain>` with your valid certificate chain when registering with {{< product-c8y-iot >}}. The header `<device certificate chain>` is not mandatory if you have an immediate issuer of the device certificate in {{< product-c8y-iot >}}.
```
Comment (saran-govindarajan) [inline content/device-integration/device-integration-rest-bundle/device-authentication.md:30]: use the name of the header - X-Ssl-Cert-Chain

Comment (cumulocitypans) [reply content/device-integration/device-integration-rest-bundle/device-authentication.md]: Done

File: content/device-integration/rest-client-examples-bundle/hello-rest-x509.md [hunk 4]
```diff
@@ -52,7 +52,7 @@ The following configuration is required before calling the device access token A
 * PLATFORM_URL - The URL of the platform.
 * PLATFORM_MTLS_PORT - Port 8443 is available for device access token API
 * DEVICE_ACCESS_TOKEN_PATH API - The endpoint responsible for the mTLS protocol.
-* LOCAL_DEVICE_CHAIN - The whole chain in PEM format.
+* LOCAL_DEVICE_CHAIN - The whole chain in PEM format. This header is not mandatory, if you have an immediate issuer of the device certificate in {{< product-c8y-iot >}}.
```
[the remark this answers is not in the document]
Comment (cumulocitypans) [reply content/device-integration/rest-client-examples-bundle/hello-rest-x509.md]: Done

File: content/device-integration/device-integration-rest-bundle/device-authentication.md [hunk 5]
```diff
@@ -13,16 +13,12 @@ Mutual TLS (mTLS) is a security protocol that uses X.509 certificates for both c
 The mTLS protocol is commonly used to secure connections in web services, APIs, and other networked applications. When generating tokens using mTLS, the process involves the authentication of both the client and the server using X.509 certificates.
 
 Retrieving device access tokens from the platform with certificates does not require the tenant ID, username and password. Authentication information will be obtained from the certificates.
+An immediate issuer of the client certificate must be provided in the platform's truststore. If this is not the case, then the whole certificate chain must send in the header and root or any intermediate certificate must be available in the platform's truststore.
```
[the remark this answers is not in the document]
Comment (cumulocitypans) [reply content/device-integration/device-integration-rest-bundle/device-authentication.md]: Done

File: content/device-integration/device-integration-rest-bundle/device-authentication.md [hunk 6]
```diff
@@ -13,16 +13,12 @@ Mutual TLS (mTLS) is a security protocol that uses X.509 certificates for both c
 The mTLS protocol is commonly used to secure connections in web services, APIs, and other networked applications. When generating tokens using mTLS, the process involves the authentication of both the client and the server using X.509 certificates.
 
 Retrieving device access tokens from the platform with certificates does not require the tenant ID, username and password. Authentication information will be obtained from the certificates.
+The device Access token can be retrieved by sending only the device leaf certificate if an immediate issuer of the device certificate is uploaded to the trusted certificates list. If the uploaded trusted certificate is not an immediate issuer of the device certificate but belongs to the device's chain of trust, then the device must send the entire certificate chain in the X-Ssl-Cert-Chain to be authenticated successfully and retrieve the device Access token.
```
Comment (BeateRixen) [inline content/device-integration/device-integration-rest-bundle/device-authentication.md:16]:
```suggestion
The device access token can be retrieved by sending only the device leaf certificate if an immediate issuer of the device certificate is uploaded to the trusted certificates list. If the uploaded trusted certificate is not an immediate issuer of the device certificate but belongs to the device's chain of trust, then the device must send the entire certificate chain in the `X-Ssl-Cert-Chain` to be authenticated successfully and retrieve the device access token.
```

File: content/device-integration/device-integration-rest-bundle/device-authentication.md [hunk 7]
```diff
@@ -31,7 +27,12 @@ The device access token API can be called by executing the following curl statem
        -X POST \
        https://<{{< product-c8y-iot >}} tenant domain>:8443/devicecontrol/deviceAccessToken
 
-Replace `<device certificate chain>` with your valid certificate chain when registering with {{< product-c8y-iot >}}.
+Replace `<device certificate chain>` with your valid certificate chain when registering with {{< product-c8y-iot >}}. Header `X-Ssl-Cert-Chain` is not mandatory if you have an immediate issuer of the device certificate in {{< product-c8y-iot >}}.
```
Comment (BeateRixen) [inline content/device-integration/device-integration-rest-bundle/device-authentication.md:30]:
```suggestion
Replace `<device certificate chain>` with your valid certificate chain when registering with {{< product-c8y-iot >}}. The header `X-Ssl-Cert-Chain` is not mandatory if you have an immediate issuer of the device certificate in {{< product-c8y-iot >}}.
```

**Review by BeateRixen — APPROVED** [review]

**Pull request merged**